### PR TITLE
Review changes: htmlElements, Date in bean, bean for Hotel, refactor Pages, ElementUtil, BookHotelTest, BookHotelPage.

### PR DESCRIPTION
**Pull request by AlesiaMianitskaya**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,6 +16,12 @@
     </properties>
 
     <dependencies>
+		<dependency>
```
Comment (DzmitryZakharenka) [inline pom.xml]: Align this block please.

Comment (DzmitryZakharenka) [inline pom.xml]: Could you tell me about htmlelements? In the lab :)

File: src/test/java/com/epam/tat/airlines/taf/bean/hotel/HotelSearchParameters.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            adultsAmount == that.adultsAmount &&
+            childrenAmount == that.childrenAmount &&
+            departureMonth == that.departureMonth &&
+            infantsAmount == that.infantsAmount &&
+            Objects.equals(destination, that.destination);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(destination, nightsAmount, roomsAmount, adultsAmount, childrenAmount, departureMonth, infantsAmount);
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/bean/hotel/HotelSearchParameters.java]: more than 80 characters in one line.

File: src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java [hunk 3]
```diff
@@ -1,33 +1,31 @@
...
 
 public class PassengerDetails {
 
     private String title;
     private String firstName;
     private String lastName;
-    private int birthdayDay;
-    private int birthdayMonth;
-    private int birthdayYear;
+    private Calendar dateOfBirth;
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java]: Maybe better name that variable like birthdayDate? It is more attractive for my opinion :)

File: src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java [hunk 4]
```diff
@@ -1,33 +1,31 @@
...
-
         this.title = title;
         this.firstName = firstName;
         this.lastName = lastName;
     }
 
-    public PassengerDetails(String firstName, String lastName, int birthdayDay,
-                            int birthdayMonth, int birthdayYear) {
-
+    public PassengerDetails(String firstName, String lastName, Calendar dateOfBirth) {
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java]: And in this case?

File: src/test/java/com/epam/tat/airlines/taf/elements/HolidaySearchForm.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.FindBy;
+import ru.yandex.qatools.htmlelements.annotations.Name;
+import ru.yandex.qatools.htmlelements.element.Button;
+import ru.yandex.qatools.htmlelements.element.HtmlElement;
+import ru.yandex.qatools.htmlelements.element.TextInput;
+
+@Name("HolidaySearchForm")
+@FindBy(xpath = "//div[@class='looking-for']//li[2]//span")
+public class HolidaySearchForm  extends HtmlElement{
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/elements/HolidaySearchForm.java]: space missed before "{"

File: src/test/java/com/epam/tat/airlines/taf/elements/VillaSearchForm.java [hunk 6]
```diff
@@ -1,108 +1,85 @@
...
-        buttonFindVillas.click();
-        return new PriceVillaPage(driver);
-    }
-
-    public void open() {
-        driver.get(PageConstant.HOME_URL);
-    }
+    private TextInput selectTo;
 
     private boolean isElementPresent(WebElement element) {
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/elements/VillaSearchForm.java]: Delete this method from here please.

File: src/test/java/com/epam/tat/airlines/taf/page/PassengerOptionPage.java [hunk 7]
```diff
@@ -65,8 +68,9 @@ public void fillChildOptions(PassengerDetails details) {
     }
 
     public void fillInfantOptions(PassengerDetails details) {
-        fillInfantName(details.getFirstName(), details.getLastName() );
-        selectInfantDOB(details.getBirthdayDay(), details.getBirthdayMonth(), details.getBirthdayYear());
+        fillInfantName(details.getFirstName(), details.getLastName());
+        selectInfantDOB(details.getDateOfBirth().get(Calendar.DAY_OF_MONTH), details.getDateOfBirth().get(Calendar.MONTH),
```
Comment (DzmitryZakharenka) [inline src/test/java/com/epam/tat/airlines/taf/page/PassengerOptionPage.java]: change please name of method like selectInfantBirthdayDay or selectInfantBirthdayDate

File: pom.xml (hunk 1 above)
Comment (AlesiaMianitskaya) [inline pom.xml]: Done

File: src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java (hunk 3 above)
Comment (AlesiaMianitskaya) [inline src/test/java/com/epam/tat/airlines/taf/bean/passenger/PassengerDetails.java]: Done

File: src/test/java/com/epam/tat/airlines/taf/bean/hotel/HotelSearchParameters.java (hunk 2 above)
Comment (AnastasiyaRahonskaya) [inline src/test/java/com/epam/tat/airlines/taf/bean/hotel/HotelSearchParameters.java]: Done

File: src/test/java/com/epam/tat/airlines/taf/elements/HolidaySearchForm.java (hunk 5 above)
Comment (AnastasiyaRahonskaya) [inline src/test/java/com/epam/tat/airlines/taf/elements/HolidaySearchForm.java]: Done

File: src/test/java/com/epam/tat/airlines/taf/page/PassengerOptionPage.java (hunk 7 above)
Comment (AlesiaMianitskaya) [inline src/test/java/com/epam/tat/airlines/taf/page/PassengerOptionPage.java]: Done

**Pull request merged**